### PR TITLE
Do not use MPI_COMM_WORLD in MG Transfer

### PR DESCRIPTION
**Pull request by kronbichler**
Related to #6016.

### CODE REVIEW

File: include/deal.II/multigrid/mg_transfer.h [hunk 1]
```diff
@@ -66,9 +66,16 @@ namespace internal
...
+        dynamic_cast<const parallel::Triangulation<DoFHandlerType::dimension,DoFHandlerType::space_dimension>*>
+        (&(dh.get_triangulation()));
+      MPI_Comm communicator = dist_tria != nullptr ?
+                              dist_tria->get_communicator() :
+                              MPI_COMM_SELF;
+
       matrix.reinit(dh.locally_owned_mg_dofs(level+1),
                     dh.locally_owned_mg_dofs(level),
-                    sp, MPI_COMM_WORLD, true);
+                    sp, communicator, true);
```
Comment (bangerth) [inline include/deal.II/multigrid/mg_transfer.h]: Good catch. This looks like it really was a bug :-)

**Comment by bangerth** [comment]
OK to merge once the tester is happy.

File: include/deal.II/multigrid/mg_transfer.h (hunk 1 above)
Comment (kronbichler) [reply include/deal.II/multigrid/mg_transfer.h]: Yes indeed.

**Pull request merged**